### PR TITLE
Feature/transaction page

### PR DESCRIPTION
**Pull request by gabito1451**
Added Transaction and Deposit page 
The user is also directed to the transaction page if already logged in

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (nwankwo-ikemefuna) [inline .vscode/settings.json:1]: Don't commit this file. It's specific to your workbench.

File: src/App.js [hunk 2]
```diff
@@ -6,14 +6,15 @@ import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
...
 import { Transaction } from "./components/transaction";
-
+import { Deposit } from "./components/deposit";
 export function App() {
   return (
     <Router>
       <Routes>
         <Route path="/register" Component={Register} />
         <Route path="/" Component={Login} />
         <Route path="/transaction" Component={Transaction} />
```
Comment (nwankwo-ikemefuna) [inline src/App.js]: The last 2 should be protected routes. You should use a higher order component (HOC) which will wrap the protected route components. Research about how to do this.

Here is a suggestion by ChatGPT:

```
import { BrowserRouter as Router, Routes, Route, Navigate } from 'react-router-dom';

// HOC for protecting routes
const ProtectedRoute = ({ component: Component, isLoggedIn, ...rest }) => {
  return (
    <Route
      {...rest}
      element={
        isLoggedIn ? (
          <Component />
        ) : (
          <Navigate to="/" replace />
        )
      }
    />
  );
};

// App component
const App = () => {
  const [isLoggedIn, setIsLoggedIn] = useState(false);

  // your login/logout logic here

  return (
    <Router>
      <Routes>
        <Route path="/register" element={<Register />} />
        <Route path="/" element={<Login setIsLoggedIn={setIsLoggedIn} />} />
        <ProtectedRoute
          path="/transaction"
          element={<Transaction />}
          isLoggedIn={isLoggedIn}
        />
        <ProtectedRoute
          path="/deposit"
          element={<Deposit />}
          isLoggedIn={isLoggedIn}
        />
      </Routes>
    </Router>
  );
};
```

File: src/components/transaction.js [hunk 3]
```diff
@@ -1,7 +1,72 @@
...
+
+  return (
+    <div>
+      <Navbar />
+      <AccountBalance />
+      <h1>Transaction History</h1>;
+      <button
+        className="btn-danger"
+        type="button"
+        onClick={() => clearTransactionHistory()}
```
Comment (nwankwo-ikemefuna) [inline src/components/transaction.js]: This is a dangerous action, you should add confirm action.

File: src/components/transaction.js [hunk 4]
```diff
@@ -1,7 +1,72 @@
...
+              <th>Type</th>
+              <th>Amount</th>
+              <th>Balance Before</th>
+              <th>Balance After</th>
+              <th>Beneficiary</th>
+            </tr>
+          </thead>
+          <tbody>
+            {userTransactionArr.map((transaction) => (
+              <tr>
```
Comment (nwankwo-ikemefuna) [inline src/components/transaction.js]: Each child in a list should have a unique "key" prop. Add key prop with transaction reference as value, as it's the only unique value of transaction.

`key={transaction.transactionReference}`

File: src/components/common-session-component/navbar.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        <Link to="/withdrawal" className="link">
+          Withdraw
+        </Link>
+        <Link to="/transfer" className="link">
+          Transfer
+        </Link>
+        <Link to="/account-profile" className="link">
+          Account Profile
+        </Link>
+        <Link to="/logout" className="link">
```
Comment (nwankwo-ikemefuna) [inline src/components/common-session-component/navbar.js]: There is no page called logout and there shouldn't be. You should attach an onClick function to the Link component that will correctly log the user out and the redirect to login page.
Something like:
```
<Link className="link" onClick={handleLogout}>
     Log out
</Link>
```

**Review by nwankwo-ikemefuna — CHANGES_REQUESTED** [review]
Looks good 👍, but needs some changes.

File: .vscode/settings.json (hunk 1 above)
Comment (gabito1451) [reply .vscode/settings.json:1]: Done

File: src/components/transaction.js (hunk 3 above)
Comment (gabito1451) [reply src/components/transaction.js]: Done

File: src/components/transaction.js (hunk 4 above)
Comment (gabito1451) [reply src/components/transaction.js]: Done

File: src/components/common-session-component/navbar.js (hunk 5 above)
Comment (gabito1451) [reply src/components/common-session-component/navbar.js]: Done

**Review by nwankwo-ikemefuna — APPROVED** [review]

**Pull request merged**